js_modules/dagit/packages/ui/src/components/TokenizingField.stories.tsx

```diff
@@ -0,0 +1,100 @@
+import {Meta} from '@storybook/react/types-6-0';
+import * as React from 'react';
+import styled from 'styled-components/macro';
+
+import {Colors} from './Colors';
+import {Group} from './Group';
+import {TokenizingField, TokenizingFieldValue} from './TokenizingField';
+
+// eslint-disable-next-line import/no-default-export
+export default {
+  title: 'TokenizingField',
+  component: TokenizingField,
+} as Meta;
+
+export const Default = () => {
+  const [value, setValue] = React.useState<TokenizingFieldValue[]>([]);
+
+  const suggestions = [
+    {
+      token: 'pipeline',
+      values: () => ['airline_demo_ingest', 'airline_demo_warehouse', 'composition'],
+    },
+    {
+      token: 'status',
+      values: () => ['QUEUED', 'NOT_STARTED', 'STARTED', 'SUCCESS', 'FAILURE', 'MANAGED'],
+    },
+  ];
+
+  return (
+    <TokenizingField
+      values={value}
+      onChange={(values) => setValue(values)}
+      suggestionProviders={suggestions}
+    />
+  );
+};
+
+export const TokenProvider = () => {
+  const [value, setValue] = React.useState<TokenizingFieldValue[]>([]);
+
+  return (
+    <TokenizingField
+      values={value}
+      onChange={(values) => setValue(values)}
+      tokens={['ben@elementl.com', 'dish@elementl.com', 'marco@elementl.com']}
+    />
+  );
+};
+
+export const TokenAndSuggestionProviders = () => {
+  const [value, setValue] = React.useState<TokenizingFieldValue[]>([]);
+
+  const suggestions = [
+    {
+      token: 'group',
+      values: () => ['core@elementl.com', 'cloud@elementl.com'],
+    },
+  ];
+
+  return (
+    <TokenizingField
+      values={value}
+      onChange={(values) => setValue(values)}
+      tokens={['ben@elementl.com', 'dish@elementl.com', 'marco@elementl.com']}
+      suggestionProviders={suggestions}
```

Comment:
I think this is nice but it's a little unclear what the behavior is when tokens / providers are passed together -- I wonder if it'd be a bit clearer if TokenizingField exposed a convenience method for building a provider in a really shorthand way for the basic tokens case? (I don't feel strongly about it though!)

Maybe this could be something like:

```

    <TokenizingField
      values={value}
      onChange={(values) => setValue(values)}
      suggestionProviders={[providerForValues(['ben@elementl.com', 'dish@elementl.com', 'marco@elementl.com'])}
/>
```

I'm a little wary of building more complexity into the TokenizingField itself because we've had a couple discussions about adding other features to it like "double-click edit an item" etc., and it could end up being a beast if it supports a lot of use cases. 

 --

Maybe we ship this as implemented for now, but if we end up making TokenizingField do more powerful stuff for the run filter bar, etc., we consider splitting this "basic non-key-value text tokens" use case out into a separate component? Curious to get your thoughts @hellendag